src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtSpoeAreasOfLawService.java

```diff
@@ -0,0 +1,106 @@
+package uk.gov.hmcts.dts.fact.services.admin;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.dts.fact.entity.Court;
+import uk.gov.hmcts.dts.fact.entity.CourtAreaOfLawSpoe;
+import uk.gov.hmcts.dts.fact.exception.DuplicatedListItemException;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw;
+import uk.gov.hmcts.dts.fact.repositories.CourtAreaOfLawSpoeRepository;
+import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
+import uk.gov.hmcts.dts.fact.util.AuditType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class AdminCourtSpoeAreasOfLawService {
+
+    private final CourtRepository courtRepository;
+    private final AdminAuditService adminAuditService;
+    private final CourtAreaOfLawSpoeRepository courtAreaOfLawSpoeRepository;
+
+    @Autowired
+    public AdminCourtSpoeAreasOfLawService(final CourtRepository courtRepository, final CourtAreaOfLawSpoeRepository courtAreaOfLawSpoeRepository, final AdminAuditService adminAuditService) {
+        this.courtRepository = courtRepository;
+        this.courtAreaOfLawSpoeRepository = courtAreaOfLawSpoeRepository;
+        this.adminAuditService = adminAuditService;
+    }
+
+    public List<AreaOfLaw> getAllSpoeAreasOfLaw() {
```

Comment:
Double checking on this one. Is it to get a complete list back to populate the select boxes on the frontend? I.e list all the areas of law, then getCourtSpoeAreasOfLawBySlug will check the ones that come back?

If so, could this use the general get areas of law endpoint that already exists?

Reply:
yes it is to get the list to populate the checkboxes on the frontend , just though would be better to separate this from court area of law list and the general endpoint returns a string list would need it to contain id aswell.